### PR TITLE
feat: reference number and and date not mandatory in Payment Entry (LAN-574)

### PR DESCRIPTION
**Pull request by melanieroedel**
First attempt by customization
Does not solve the problem discussed here: https://discuss.erpnext.com/t/disabling-mandatory-field-reference-no-in-payment-entry/63490/2

### CODE REVIEW

File: landa/landa_sales/custom/payment_entry.json [hunk 1]
```diff
@@ -347,6 +347,102 @@
...
+   "name": "Payment Entry-reference_no-mandatory_depends_on",
+   "owner": "Administrator",
+   "parent": null,
+   "parentfield": null,
+   "parenttype": null,
+   "property": "mandatory_depends_on",
+   "property_type": "Data",
+   "row_name": null,
+   "value": ""
+  },
```
Comment (barredterra) [inline landa/landa_sales/custom/payment_entry.json]:
```suggestion
```

**Review by barredterra — DISMISSED** [review]
We don't need these property setters anymore, now that we have the default values, right?

File: landa/landa_sales/custom/payment_entry.json (hunk 1 above)
Comment (melanieroedel) [reply landa/landa_sales/custom/payment_entry.json]: I think we might. There is a strange double functionality as pointed out in the screenshots in the ticket.
Part of it is from this, part of it from an additional json file. So I decided to leave it in.
If you think we should take it out, feel free to delete it though!

**Review by scdanieli — APPROVED** [review]

**Pull request merged**